x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpClient.java

```diff
@@ -145,12 +169,14 @@ private void respondUsingUtilityThread(HttpResponse response, HttpRequest reques
                     e
                 );
                 listener.onFailure(e);
+            } finally {
+                EntityUtils.consumeQuietly(response.getEntity());
             }
         });
     }
 
     private void failUsingUtilityThread(Exception exception, ActionListener<?> listener) {
```

Comment:
This method name should be updated to reflect the fact that it no longer uses the utility thread pool.